main.go

```diff
@@ -196,6 +173,7 @@ func main() {
 	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
 	<-signalChan
 	log.Println("Shutdown signal received, exiting...")
+	p.db.Close()
```

Comment:
Consider using a `defer` for this right after you open the db connection.

Reply:
If I add a defer right after the db connection that connection will be close and no operation will be able to be preformed. I can either open and close the connection for each call or I can keep the connection open as advised https://github.com/jinzhu/gorm/issues/1427. This placement will keep the connection open until the shutdown signal is received.

Reply:
defer does not execute until main() returns.  It's the idiomatic way in go to deal with cleaning up resources.  See https://tour.golang.org/flowcontrol/12

Reply:
Yes that is true but because I broke out the db operation into functions and pass around the pointer to the db, I think main thinks there is nothing to do so it closes the connection. I tried a tests and it does close the connection.
`(sql: database is closed) 
[2018-11-09 14:28:56]  
`

Reply:
Maybe I'm missing something... I have it in the db method... maybe I need to create a main function and put the open/close there?

Reply:
Defered calls are only executed when the enclosing function is done, if the defer on the db close from main isn't working then something funky is going on.  I can take a look next week, but it's worrisome that the defer isn't working.

Reply:
Well it does work if I put all operation in the same function or in main. I'll mess around with it. We can have a hangout on Monday and go over it.